lib/ruby_lsp/ruby_lsp_rails/server.rb

```diff
@@ -79,6 +81,33 @@ def execute(request, params)
 
       private
 
+      sig { params(name: String).returns(T::Hash[Symbol, T.untyped]) }
+      def route_location(name)
+        # Older versions of Rails don't support `route_source_locations`.
+        # We also check it hasn't been disabled.
+        unless ActionDispatch::Routing::Mapper.respond_to?(:route_source_locations) &&
+            ActionDispatch::Routing::Mapper.route_source_locations
```

Comment:
We can't rely purely on the version check because the app could have an initializer which sets `route_source_locations` to false.

Reply:
I think for this first iteration I'd like to leave the approach as-is, and we can re-visit.

Reply:
If that's the case, then `ActionDispatch::Routing::Mapper.route_source_locations` needs to be in `server.rb` while `ActionDispatch::Routing::Mapper.respond_to?(:route_source_locations)` can be on the addon side. This is because only `server.rb` would be run with the application booted and initializers executed.

However, I also feel like it's not a user-facing option. Otherwise, it'd be accessible via `config` as well. From the [search result](https://github.com/search?q=repo%3Arails%2Frails%20route_source_locations&type=code), I feel that it's an internal option that's only controlled by Rails environment. If that's true, though, perhaps we don't even check its value as we'd always be running in dev environment anyway? 🤔